src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/IHeartbeatProvider.cs

```diff
@@ -4,10 +4,22 @@
     using System.Collections.Generic;
     using Microsoft.ApplicationInsights.Extensibility;
 
-    internal interface IHeartbeatProvider : IHeartbeatPropertyManager, IDisposable
+    internal interface IHeartbeatProvider : IDisposable
```

Comment:
Bit confused with so many interfaces.. IHeartbeatProvider has Add/Set property methods.. But its also duplicated in IHeartbeatPropertyManager. I think we should not have this duplication unless there is a strong reason to it.

Reply:
Will make some comments.